lib/ClientGeneratorFromXml.php

```diff
@@ -115,7 +115,8 @@ protected function shouldIncludeType($type)
 			return false;
 
 		$type = strval($type);
-		return !count($this->_includeTypes) || isset($this->_includeTypes[$type]);
+		return ((!is_array($this->_includeTypes) && !$this->_includeTypes instanceof Countable)  || !count($this->_includeTypes) || isset($this->_includeTypes[$type]));
```

Comment:
Since _includeTypes is defined as array , would not it be enough to check is_array?
!is_array($this->_includeTypes) || !count($this->_includeTypes) || isset($this->_includeTypes[$type])

Reply:
In that case it seems that is_array is redundant.

Reply:
@MosheMaorKaltura - indeed, $this->_includeTypes is an array, not an object and therefore, at the moment, checking for `is_array($this->_includeTypes)` is enough. However, it doesn't hurt to check for both. I added the same check in several places. It's stupid that the SPL [Standard PHP Library] does not have an is_countable() method but it doesn't...
@tan-tan-kanarek - I'm a little confused about your comment:) Did you mean the `$this->_includeTypes instanceof Countable` is redundant? the is_array() is needed, otherwise, with PHP 7.2, you'll get:
>  fix PHP Warning: count(): Parameter must be an array or an object that implements Countable in /opt/kaltura/clients-generator/lib/ClientGeneratorFromXml.php on line 119

Reply:
Sorry, I'm not familiar enough with 7.2, I thought that array implements Countable.